cache/manager.go

```diff
@@ -104,13 +110,39 @@ func NewManager(opt ManagerOpt) (Manager, error) {
 		Applier:         opt.Applier,
 		Differ:          opt.Differ,
 		MetadataStore:   opt.MetadataStore,
+		mountPoolRoot:   opt.MountPoolRoot,
 		records:         make(map[string]*cacheRecord),
 	}
 
 	if err := cm.init(context.TODO()); err != nil {
 		return nil, err
 	}
 
+	if opt.MountPoolRoot != "" {
+		if err := os.MkdirAll(opt.MountPoolRoot, 0700); err != nil {
+			return nil, fmt.Errorf("failed to prepare mount pool: %w", err)
+		}
+		// If MountPoolRoot is specified, remove existing mounts to avoid conflict.
+		files, err := os.ReadDir(opt.MountPoolRoot)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read mount pool: %w", err)
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				dir := filepath.Join(opt.MountPoolRoot, file.Name())
+				logrus.Debugf("cleaning up existing temporary mount %q", dir)
+				if err := mount.Unmount(dir, 0); err != nil {
+					logrus.WithError(err).Warnf("failed to unmount existing temporary mount %q", dir)
```

Comment:
I guess if the host experienced a sudden power off then when buildkitd starts up again you could end up with dirs left here but no mounts. In this case `Unmount` would always fail but since we `continue` the dir would never be removed and this log would be printed every startup forever. Maybe it's better to check the err and try removing the dir if the error indicates there wasn't a mount there.

Also, nit about using `bklog` for these statements.

Reply:
Thank you for the review.

> I guess if the host experienced a sudden power off then when buildkitd starts up again you could end up with dirs left here but no mounts. In this case Unmount would always fail but since we continue the dir would never be removed and this log would be printed every startup forever.

containerd's `mount.Unmount` ignores `EINVAL` which can is returned when the dir isn't a mount point. So I think it won't be occur.

> Also, nit about using bklog for these statements.

Fixed to use bklog.

> Also, if it's feasible it would be nice to have a test of this behavior. Unfortunately I don't think the client integ tests would support something like sending SIGKILL to buildkitd and then restarting it, but even something in the unit tests would be helpful.

Thank you for the suggestion.
Added an unit test.